shanoir-ng-users/src/main/java/org/shanoir/ng/extensionrequest/controller/ExtensionRequestApiController.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.extensionrequest.controller;
+
+import org.shanoir.ng.email.EmailService;
+import org.shanoir.ng.extensionrequest.model.ExtensionRequestInfo;
+import org.shanoir.ng.shared.controller.AbstractUserRequestApiController;
+import org.shanoir.ng.shared.exception.EntityNotFoundException;
+import org.shanoir.ng.shared.exception.ShanoirException;
+import org.shanoir.ng.user.model.User;
+import org.shanoir.ng.user.utils.KeycloakClient;
+import org.shanoir.ng.utils.SecurityContextUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+
+@Controller
+public class ExtensionRequestApiController extends AbstractUserRequestApiController implements ExtensionRequestApi {
+
+	private static final Logger LOG = LoggerFactory.getLogger(ExtensionRequestApiController.class);
+
+	@Autowired
+	private KeycloakClient keycloakClient;
+
+	@Autowired
+	private EmailService emailService;
+
+	@Override
+	public ResponseEntity<Void> requestExtension(@RequestBody final ExtensionRequestInfo requestInfo) {
+		try {
+			SecurityContextUtil.initAuthenticationContext("ADMIN_ROLE");
```

Comment:
Hi Jean-Come, could we do it without that admin_role sec_context? From my point of view, everything down to the service methods, should be open for access. I think it is for security reasons more dangerous to enable an admin_access here then allowing an outsider to ask for an account extension for a given account, especially as the account remains disabled until the extension request is accepted. With kind regards, Michael

Reply:
open for access for this request